### ISSUE TITLE
FrontPage refresh

### ISSUE THREAD

**Issue by SjoenH** [equinor/MAD-VSM-WEB#113]
https://www.figma.com/file/IkHwmIQrsT0iR34f5R5UnZ/VSM?node-id=2552%3A89145

**Comment by Magos** [comment]
Da har jeg et draft av migreringsscript til ny databasestruktur for å støtte dette. Innholdet er:

- Først, litt renhold - fjerning av kolonner som ikke er i bruk i modellen vår. For å gjennomføre dette vil det være slettede data på noen nivåer; det handler om <10 rows i hvert nivå jeg har undersøkt og jeg antar alle er opprettet via SQL heller enn applikasjonen. I prodnivået er det 1 påvirket row som gjelder @pagodo. 
- Deling av nåværende UserAccesses i to tabeller User og Accesses. Dette hjelper oss å holde bruker-lagringen på en god normalform.
- Introduksjon av ny tabell UserToVSM, som linker mellom User og VSM for å holde informasjon om Favorite-funksjonen. I tillegg har den en Created-timestamp.
- Introduksjon av et nytt view UserAccesses som skal erstatte den gamle tabellen. Dette kommer som eget script da SQLMS hevder CREATE VIEW må være først i sin batch, og det avhenger av at de gamle navnene er ryddet bort før det kan kjøres.


[Favorites_draft_1.zip](https://github.com/equinor/MAD-VSM-WEB/files/7144187/Favorites_draft_1.zip)

**Comment by SjoenH** [comment]
@Magos, I talked with Line. Search for a specific person should give back all project where that user has edit-access. That means an owner, admin, or contributor role.

**Comment by martabenthem** [comment]
@Magos, we believe that in order to faving and unfaving a project, we need a PATCH method. 

Right now, we are using a POST method that expects a VSM project, where the VSM projects's ID is provided. There are two issues with that:

- We are sending the entire VSM project by faving/unfaving - if someone else were working on that project at the same time as we made the "fave" request, we would be overriding their changes.
- We cannot fave/unfave VSM projects we don't have edit access to.

**Comment by Magos** [comment]
Looking at a separate "fave" endpoint now.

**Comment by Magos** [comment]
Update since last

- Separate fav/unfav endpoint now exists on /api/v{api-version}/project/favorite/{id} (PUT favs, DELETE unfavs)
- Corrected a situation where new users could be stored multiple times, damaging data integrity wrt. favorites.
- Corrected a situation where new user access rule (sharing a VSM with someone) failed to save.
- Corrected a situation where new projects (empty or duplicates) failed to save.

**Comment by SjoenH** [comment]
@ntorsen Ready for testing in dev environment. https://web-vsm-dev.radix.equinor.com/
Add a comment if you find something strange. If all is good, you may promote it to QA environment and notify our POs.

**Comment by ntorsen** [comment]
**Results from testing 29.09**
![image](https://user-images.githubusercontent.com/89014017/135276926-174ea8c5-311e-43ad-ae9a-bc41a00e5f21.png)

**Comment by SjoenH** [comment]
Thanks for testing @ntorsen 
Not a big thing but, I would appreciate it if next time you add the images and text directly into GitHub so that it is easier to read and respond to your feedback 😄 


**Search for name & users:**
> Able to search for username (such as ntor, magsa, etc) and name of project.
> - Are we going to add an option to search for a specific person where you type in first name/last name. Maybe it will come in handy if you don't know the username?

Yes, but full-name-search is dependent upon #148. Let's add it later. We only support short-name-search for now.


**Alphabetically:**
> Why does Untitled VSM come first up for 'All projects'?

We display your own `Untitled` processes, but should hide other people's `Untitled` processes. Let me know if it doesn't follow this rule. :)


> Should it come here at all? I was expecting it to only land on 'My projects".

The section is all projects, not everyone else's projects. So I think it makes sense to show your projects in line with the other's?

**Search**
> If it exists an `Untitled VSM` for `All projects`, why am I not able to search for it? Tried both searching for `Untitled` and `VSM`

Any `Untitled VSM` does not have a given name, thus it is untitled or rather not set in the back-end. But you make a valid point that searching for `Untitled VSMs` might come in handy. 🤔 


**Deleting a vsm**
> Responding "weirdly" when I try to delete a vsm (both from favorites and all projects):

Nice catch! 💪  @Magos could you have a look at this? Summary: We have a conflict when trying to delete a project that is faved.
![image](https://user-images.githubusercontent.com/3164065/135391189-bef15f2f-d77c-4259-8133-c25b7067a706.png)

> It should take me to the Main page after deleting it, but now I get "An error occurred on client, and problems with logging me in when clicking on 'Flyt' on the left-hand side.

Yes, would be covered by the `deleting a favourite VSM bug`-fix.

**Comment by ntorsen** [comment]
> We display your own `Untitled` processes, but should hide other people's `Untitled` processes. Let me know if it doesn't follow this rule. :)

Yes, it follows this rule 👍 However I get other people's duplicate of untitled vsm. Is this an exception to rule? @SjoenH 
![image](https://user-images.githubusercontent.com/89014017/135395118-44c09edf-d353-4c03-a4e9-dae68f640028.png)

> The section is all projects, not everyone else's projects. So I think it makes sense to show your projects in line with the other's?
Yes that makes sense 💯

**Comment by SjoenH** [comment]
> Yes, it follows this rule 👍 However I get other people's duplicate of untitled vsm. Is this an exception to rule?

Yes. I don't think it will be a common thing people will do in prod. 🤔 Let's do something about it if it becomes a problem.

**Comment by ntorsen** [comment]
@Magos will you let me know when the fix for deleting vsm is put in?

**Comment by Magos** [comment]
@ntorsen The fix was updating an SQL Stored Procedure, so no deploy for this - Dev environment is updated now.

**Comment by ntorsen** [comment]
Change message 500 for deleted projects @Magos 
![image](https://user-images.githubusercontent.com/89014017/135423175-5f042550-2b89-4483-9e1e-072c6b9b7f79.png)

**Comment by Magos** [comment]
@ntorsen @Sjoen 404 response for nonexistent project is now deployed to Dev.
![firefox_icJ2tQ3EdV](https://user-images.githubusercontent.com/1004386/135439116-f362d076-9e40-468e-9d58-b3fd25e581fc.png)
H

**Comment by ntorsen** [comment]
Looking good 👍 
I'm promoting DEV into QA

**Comment by SjoenH** [comment]
@equinor/flyt-product-owners This is now ready for Testing in QA enviroment. https://web-vsm-qa.radix.equinor.com

Let us know if you have any concerns. If not, we will prepare for a release.
@Magos we will need to look into a good timing for release to Production. Expected downtime approx 10 minutes.

**Comment by knudsvik** [comment]
I have tested the QA version, nice improvements! Do have some comments:
- The search bar should be available in all screens (all projects, your projects, your favourite projects), the separate search page is not needed 
- When many participants are active in one project, the icons looks a bit weird. ![image](https://user-images.githubusercontent.com/25986427/135574472-e5d97194-29e8-4fde-af5d-f209360b9512.png) Is it possible to this any cleaner? E.g. having the owner as a separate, then stacking admins and collaborators or something? I guess the layout of the cards will have to change with #159 , so could be adressed there.
- Both in Edge and Chrome there is an issue with the width of the site, it leaves room for scrolling sideways 0.5 cm.. This shows on all pages except for the favourite page.
- Discrepencies between the title of the pages and the text when hovering the side bar: "My projects" vs "Your projects" and "Favourite" vs "Your favourite projects"
- In the Figma design, the search bar had additional functionality, like searching for several things, is this a separate issue now for future dev?

**Comment by SjoenH** [comment]
@knudsvik thanks for the feedback.
> * The search bar should be available in all screens (all projects, your projects, your favourite projects), the separate search page is not needed

Ok, then the search would be affected by your choice in the left-navbar. All projects + search would give the same as the search-page atm.

> * When many participants are active in one project, the icons looks a bit weird. ![image](https://user-images.githubusercontent.com/25986427/135574472-e5d97194-29e8-4fde-af5d-f209360b9512.png) Is it possible to this any cleaner? E.g. having the owner as a separate, then stacking admins and collaborators or something? I guess the layout of the cards will have to change with [Project labels #159](https://github.com/equinor/MAD-VSM-WEB/issues/159) , so could be adressed there.

Yes, I see your point. @Linedueo could you keep this in mind when doing #159 ? Also, I think the cards are a bit short with the new design. If for instance, you have a long title, it will now be cut off really early.

> * Both in Edge and Chrome there is an issue with the width of the site, it leaves room for scrolling sideways 0.5 cm.. This shows on all pages except for the favourite page.

@martabenthem let's frontPage container styling to
```css
width: calc(100% - 70px);
padding-left: 70px;
```

> * Discrepencies between the title of the pages and the text when hovering the side bar: "My projects" vs "Your projects" and "Favourite" vs "Your favourite projects"

@martabenthem Please make it follow EDS guidelines ref https://github.com/equinor/design-system/blob/documentation/apps/storefront/docs/guidelines/accessibility/overview.mdx#call-to-actions
![image](https://user-images.githubusercontent.com/3164065/135582797-031f6fec-5525-4d06-b9a8-0c9d0eac05df.png)

> * In the Figma design, the search bar had additional functionality, like searching for several things, is this a separate issue now for future dev?

Right now the search support shortnames of people with access to a process and the process title. We could probably make this more intuitive. @Linedueo 
For full-name search we are still waiting on #148.

----
One last thing, we are not consistent on the naming `projects` vs `processes`. Should we stick to one, or is it fine to use both?

**Comment by knudsvik** [comment]
> One last thing, we are not consistent on the naming `projects` vs `processes`. Should we stick to one, or is it fine to use both?

Was thinking the same. I think that `Processses` are better than `Projects`, but harder to pronounce. The url's (e.g. https://flyt.equinor.com/projects/475) already says project.. don't want to ruin old links, but maybe this could be fixed in a smart way too. I'll check with @askbulle about this.

**Comment by knudsvik** [comment]
@askbulle : definitivt `Processes`! Good catch

**Comment by SjoenH** [comment]
@knudsvik 
1. OK, changing to **processes** (ping @martabenthem)
2. Yes, I think so. I'll look into setting up a redirect for `/project` into `/process`.

**Comment by martabenthem** [comment]
@SjoenH I have now adressed  However:

- I have changed "Projects" to "Process" everywhere except in the URL routing. I will wait to discuss this with you before redirecting.

- There is still a small bug concerning the pagination in the "All processes" page (again, the pagination bar shows even when the search returns a couple of cards).

**Comment by knudsvik** [comment]
Can you also please change `Last created` to `Date created`

![image](https://user-images.githubusercontent.com/25986427/135808801-b9b686e5-ee6f-460c-928d-608ff9ab8c7f.png)

**Comment by martabenthem** [comment]
@knudsvik that can be done! I guess also `Last modified` to `Date modified`?

**Comment by SjoenH** [comment]
Right now, we only show ~15 processes on the front-page. This looks fine most of the time.
But we could calculate how many we can show (height and width) and display as many as possible on a page. #218

@knudsvik Let us know if this should be fixed before we release or if it is not prioritized. Created an issue for it, so we may do it later.

**Comment by SjoenH** [comment]
@ntorsen ready for testing in Dev. https://web-vsm-dev.radix.equinor.com

**Comment by SjoenH** [comment]
@ntorsen 
Note, that this front-page change does not include the risk-feature #155 atm. So you might see a weird rendering like this 
![image](https://user-images.githubusercontent.com/3164065/135859898-72985620-ee8b-448b-9af9-da769fa88a66.png)
Apart from that, everything else should work as expected.

**Comment by Linedueo** [comment]
@SjoenH Front page design justeringer- venstre justert search bar og tittel, flush med cards. searchbar og tittel flyter litt nå. cards skal være venstrestilt og ikke sentrert. Pagination skal være høyrestilt. Det hadde også vært fint og fått sentrert logo over ikoner i left nav. Vi bruker grønn logo. Det er også fryktelig mye spacing mellom logo og navn.

**Comment by Linedueo** [comment]
Se figma skisser

**Comment by SjoenH** [comment]
Thanks for the feedback @Linedueo. I created a new issue based on it  #220 
I don't consider these improvements a blocker for the release of these new features.

So instead of stopping our testing now, I'm suggesting we do these changes asap after this. We also have #159 that would need more front-end changes.
@equinor/flyt-product-owners  let me know if you think different about this.

**Comment by ntorsen** [comment]
**Testing went good 🥇 . Here are some feedback from testing:**
- My old bookmark does not work, I get an error 404. 
![image](https://user-images.githubusercontent.com/89014017/135992222-1d51db31-895f-412b-8dd6-48310f645b9e.png)
![image](https://user-images.githubusercontent.com/89014017/135992283-98ec3947-a590-467e-8016-ebd133930bf1.png)

Does not work when refreshing site, but works when clicking on "Flyt". So there is a workaround for this. 
No problem searching for old processes by using projects/numbers in the url 💯 

- Something else that I want to mention is that when you either delete or duplicate a process from 'My favorites' or 'My processes' you will automatically land on 'All processes'. So it its set as a default to land on this page when doing these alterations. Is this something worth looking into in the future?

**Comment by SjoenH** [comment]
Thanks for testing. I don't think that is an issue. Moving this to QA now.

**Comment by SjoenH** [comment]
@knudsvik & @askbulle 🚀 
Ready for UAT in QA-environment: https://web-vsm-qa.radix.equinor.com/

**Comment by knudsvik** [comment]
Looks good! You may release 🥳

**Comment by knudsvik** [comment]
This bar chart shows at which hour of day the changes has happened, that should be a good indication of when it is suitable to do this

![image](https://user-images.githubusercontent.com/25986427/136160163-261a5f5c-409a-4950-a6eb-cd4e74b76f0c.png)

**Comment by SjoenH** [comment]
### I suggest releasing at 08:00 tomorrow 7. Oct.

**I suggest at 8 tomorrow**. _Which is still quite early with not that much activity._
If this is ok with you: - It would be nice to inform the users about some downtime. What is the best channel for this? Could you @knudsvik relay the message? :)

### For future changes:
We probably should think about adding some service-message banner so that we can schedule downtime etc: #224 
This is something we have for our other apps in the team, so shouldn't take too long to implement.

---
### Notes about that graph
Correct me if I'm wrong. Trying to wrap my head around this... 😅 

This graph is considering all the pkChangeLog items, and does not necessarily mean that we get a lot of changes every day in that timeframe. It could just be a lot of changes done in that time on one day... 🤔 Anyway, I agree that it probably is a good indication on when the tool is used overall.

We took a look in the DB ourselves and noted that the above is in UTC time. So add 2 hours for Norway. 
The ideal times for downtime are [...,08:00] or [17:00, 20:00] or [22:00,...], but that is outside our Workday... (Kjernetid 9→15)

**Comment by knudsvik** [comment]
Good catch wih the UTC! These are the sum of pkChangeLog items, not an average so it could probably improve. But still at good indication of when it is used. We don't have a proper channel to communicate it in, hence I like the #224 . Just release it asap tomorrow morning!

**Comment by SjoenH** [comment]
Super. 💪 Will do 🚀

**Comment by SjoenH** [comment]
QA build ready to promote after API tomorrow at 08:00.
[Press here to promote QA to PROD](https://console.radix.equinor.com/applications/vsm/jobs/new?pipeline=promote&deploymentName=qa-ygkgc-l2vsobtj&fromEnvironment=qa) 👈 🚀

**Comment by SjoenH** [comment]
API release in progress.
![image](https://user-images.githubusercontent.com/3164065/136329614-5785261f-1ee0-4901-b8fe-7944e7bcf041.png)

**Comment by SjoenH** [comment]
**Note:** We are having some issues with fave-ing processes. @Magos is looking into it now.

**Comment by knudsvik** [comment]
Not sure if it's a bug or intentional but I wanted to search for some processes that have "P&I" in the title. I did not get any results, not with brackets either. I guess it's the "&" somehow making problems? This is the same search result from the power BI report:

![image](https://user-images.githubusercontent.com/25986427/136333749-f2f07323-21f6-42fc-a2b8-defa632a3dcc.png)
